src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.dsde.firecloud.dataaccess
+
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.elasticsearch.search.aggregations.{AggregationBuilders, Aggregations}
+import org.elasticsearch.client.transport.TransportClient
+import org.elasticsearch.action.search.{SearchRequest, SearchRequestBuilder, SearchResponse}
+import org.elasticsearch.search.aggregations.bucket.terms.Terms
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
+
+  def createListOfMusts(fields: Map[String, Seq[String]]): Seq[QueryMap] = {
+    (fields map {
+      case (k, v) => ESBool(createESShouldForTerms(k, v))
+    }).toSeq
+  }
+
+  def createESShouldForTerms(attribute: String, terms: Seq[String]): ESShould = {
+    val clauses: Seq[ESTerm] = terms map {
+      case (term: String) => ESTerm(Map(attribute -> term))
+    }
+    ESShould(clauses)
+  }
+
+  def createQuery(criteria: LibrarySearchParams): Seq[QueryMap] = {
+    (criteria.searchString, criteria.searchFields.size) match {
+      case (None | Some(""), 0) => Seq(new ESMatchAll)
+      case (None | Some(""), _) => createListOfMusts(criteria.searchFields)
+      case (Some(searchTerm: String), 0) => Seq(new ESMatch(searchTerm.toLowerCase))
+      case (Some(searchTerm: String), _) => createListOfMusts(criteria.searchFields) :+ new ESMatch(searchTerm.toLowerCase)
+    }
```

Comment:
well, it doesn't create the correct query when the searchString is empty but there are search facets. it adds a match_all section to the query... i would really like to be able to reason about these separately... 

how about this?
```  def createQuery(criteria: LibrarySearchParams): Seq[QueryMap] = {

    val facetNav: Seq[QueryMap] = criteria.searchFields match {
      case navs if navs.isEmpty => Seq.empty
      case navs => createListOfMusts(navs)
    }

    val search: Option[QueryMap] = criteria.searchString match {
      case None => None
      case Some(searchTerm) if searchTerm.trim == "" => None
      case Some(searchTerm) => Some(new ESMatch(searchTerm))
    }

    if (facetNav.isEmpty)
      Seq(new ESMatchAll)
    else if (search.isDefined)
      facetNav :+ search.get
    else
      facetNav
  }
```